apps/common/hooks/useBalances.tsx

```diff
@@ -181,8 +180,14 @@ export function	useBalances(props?: TUseBalancesReq): TUseBalancesRes {
 		data.current[workerForChainID].nonce += 1;
 
 		performBatchedUpdates((): void => {
+			set_balances((b): TNDict<TDict<TBalanceData>> => ({
+				...b,
+				[workerForChainID]: {
+					...(b[workerForChainID] || {}),
+					...data.current[workerForChainID].balances
+				}
```

Comment:
Wonder if we can extract this logic elsewhere, it's repeated in a bunch of places